openwebstart/src/main/java/com/openwebstart/proxy/pac/PacFileEvaluator.java

```diff
@@ -116,6 +113,27 @@ String getProxies(final URI uri) {
         if (cachedResult != null) {
             return cachedResult;
         }
+        
+        // pacUrl.openStream() must not be called in constructor, otherwise it will initialize HttpURLConnection.userAgent
+        // before System.setProperty(HTTP_AGENT, httpAgent) is called in net.sourceforge.jnlp.JNLPFile
+        if (pacContents == null) {
+           LOG.debug("Open PAC url '{}'", pacUrl);
+           // need to temporarly deactivate ProxySelector, otherwise pacUrl.openStream() will call getProxies and throw StackOverflowError after a while
+           ProxySelector proxySelector = ProxySelector.getDefault();
+           ProxySelector.setDefault(null);
+           try (final InputStream inputStream = pacUrl.openStream()) {
+              //PAC supports ASCII and new versions support UTF-8 -> https://en.wikipedia.org/wiki/Proxy_auto-config#PAC_Character-Encoding
+              pacContents = IOUtils.readContentAsUtf8String(inputStream);
+           }
+           catch (IOException e) {
+              LOG.warn("Cannot open PAC url '{}' due to {}", pacUrl, e.getMessage());
```

Comment:
instead of warning here we must terminate the application:

```
            DialogFactory.showErrorDialog(Translator.getInstance().translate("proxy.error.creationFailed"), e);
            return JNLPRuntime.exit(-1);
```